cmd/setup-etcd-environment/run.go

```diff
@@ -193,70 +257,51 @@ func setExportEnv(etcdName, dns string) (map[string]string, error) {
 			glog.Errorf("no members found in member-config")
 			return false, nil
 		}
-		var memberList []string
 		for _, m := range members {
 			memberList = append(memberList, fmt.Sprintf("%s=%s", m.Name, m.PeerURLS[0]))
 		}
-		memberList = append(memberList, fmt.Sprintf("%s=https://%s:2380", etcdName, dns))
-		exportEnv["INITIAL_CLUSTER"] = strings.Join(memberList, ",")
-		exportEnv["INITIAL_CLUSTER_STATE"] = etcdInitialExisting
+		memberList = append(memberList, fmt.Sprintf("%s=https://%s:2380", s.etcdName, s.etcdDNS))
 		return true, nil
 	})
-
-	ep, err := client.CoreV1().Endpoints("openshift-etcd").Get("etcd", metav1.GetOptions{})
-	if err != nil {
-		return nil, err
-	}
-	hostEtcdEndpoint, err := client.CoreV1().Endpoints("openshift-etcd").Get("host-etcd", metav1.GetOptions{})
-	if err != nil {
-		return nil, err
-	}
-	if len(hostEtcdEndpoint.Subsets) != 1 {
-		return nil, fmt.Errorf("openshift-etcd/host-etcd endpoint subset length should be %d, found %d", 1, len(hostEtcdEndpoint.Subsets))
-	}
-
-	endpoints := make([]string, 0)
-	for _, member := range hostEtcdEndpoint.Subsets[0].Addresses {
-		if member.Hostname == "etcd-bootstrap" {
-			endpoints = append(endpoints, "https://etcd-bootstrap."+runOpts.discoverySRV+":2379")
-			break
-		}
-	}
-	if len(ep.Subsets) != 1 {
-		return nil, fmt.Errorf("openshift-etcd/etcd endpoint subset length should be %d, found %d", 1, len(ep.Subsets))
-	}
-	for _, s := range ep.Subsets[0].Addresses {
-		endpoints = append(endpoints, "https://"+s.IP+":2379")
-	}
-
-	exportEnv["ENDPOINTS"] = strings.Join(endpoints, ",")
-	return exportEnv, nil
+	return memberList
 }
 
-// setBootstrapEnv populates and returns a map based on envs from file.
-func setBootstrapEnv(envFile, discoverySRV string, bootstrapSRV bool) (map[string]string, error) {
+// setRecoveryEnv populates ENV variables used for recovery.
+func (s *SetupEnv) getBootstrapEnv() (map[string]string, error) {
 	bootstrapEnv := make(map[string]string)
-	if _, err := os.Stat(envFile); !os.IsNotExist(err) {
-		recoveryEnv, err := godotenv.Read(envFile)
+	if _, err := os.Stat(s.opts.outputFile); !os.IsNotExist(err) {
+		existingEnv, err := godotenv.Read(s.opts.outputFile)
 		if err != nil {
 			return nil, err
 		}
 		// persist any observed envs used for recovery
 		for _, val := range []string{"INITIAL_CLUSTER", "INITIAL_CLUSTER_STATE"} {
 			rkey := fmt.Sprintf("ETCD_%s", val)
-			if recoveryEnv[rkey] != "" {
-				bootstrapEnv[val] = recoveryEnv[rkey]
+			if existingEnv[rkey] != "" {
```

Comment:
Curious at what is happening here with respect to this env variables?

In all test cases we test for `""` value of these variables, but we still set a non-nil values, this is getting me confused.

Would it make sense to have a comment to describe what value is expected in normal bootstrap and disaster recovery scenarios for better readability and debugging?

Reply:
I think I can document this better, sure.

Reply:
@retroflexer do you want to comment on the test cases?

Reply:
@alaypatel07 here is my understanding. With CEO added, now we have three different scenarios where we are setting up the export environment:
1. Normal bootstrap scenario: we should always have `DISCOVERY_SRV` 
2. Disaster Recovery case: we want to retain `INITIAL_CLUSTER` and `INITIAL_CLUSTER_STATE`. Also we need to make sure we do not include `DISCOVERY_SRV`
3. CEO is scaling: We make sure we retain `INITIAL_CLUSTER` and `INITIAL_CLUSTER_STATE`; also build the `ENDPOINTS` by querying.

Reply:
I agree with @retroflexer, I think the logic is strait forward enough if we need to handle any additional cases down the road we can revisit. Marking resolved